kivy/garden/__init__.py

```diff
@@ -151,6 +152,12 @@
     garden_app_dir = join(realpath(sys._MEIPASS), 'libs', 'garden')
 else:
     garden_app_dir = join(realpath(dirname(sys.argv[0])), 'libs', 'garden')
+#: Fixes issue #4030 in kivy where garden path is incorrect on iOS
+if platform == "ios":
+    from os.path import join, dirname
+    import __main__
+    main_py_file = __main__.__file__
```

Comment:
Why can't we use `import kivy.garden`? I'm weary about `import __main__`  because I'm not sure if this will always work.

Reply:
Not sure exactly what you mean, could you clarify with an example maybe? The fix I suggest is from Tito's answer here: https://github.com/kivy-garden/garden.mapview/issues/7#issuecomment-188471962

Reply:
Yes, but you changed tito's code by adding `import __main__` and I'm not sure that's correct. Could you not use the identical code that tito posted or was there a reason you added you're changing it?

Reply:
Ah. The goal is to reference the folder where `main.py` lives, because that's where the `lib` folder should be installed. (Assuming `main.py` is directly inside your project folder, and the user runs `garden install --app flower_name` from inside that project folder)

In Tito's example, that code is to be put in your `main.py` file, so `dirname(__file__)` refers to the directory that `main.py` is in. 
To make this a general fix, I needed to replace `__file__` with something that referred to the `main.py` file being run.

Perhaps there's a better way to refer to the folder where `main.py` lives?

Reply:
Oh, ok, I understand. For some reason I thought this was an issue when it's installed with `garden install --kivy flower_name`. Should've read the text more carefully.

I'm assuming you tested this on ios and it works?

Reply:
Yep, it worked for me on iOS.